File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/server/SilenceInfoPacket.java
package pe.nanamochi.banchus.packets.server;

import java.io.IOException;
import java.io.OutputStream;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.core.ServerPacket;
import pe.nanamochi.banchus.io.IDataWriter;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SilenceInfoPacket implements ServerPacket {
  private int silenceLength;

  @Override
  public Packets getPacketType() {
    return Packets.BANCHO_SILENCE_INFO;
  }

  @Override
  public void write(IDataWriter writer, OutputStream stream) throws IOException {
    writer.writeInt32(stream, silenceLength);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/server/SilenceInfoPacket.kt
package pe.nanamochi.banchus.packets.server

import java.io.OutputStream
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.core.ServerPacket
import pe.nanamochi.banchus.io.DataWriter

class SilenceInfoPacket(var silenceLength: Int = 0) : ServerPacket() {
    override val type = PacketType.BANCHO_SILENCE_INFO

    override fun write(writer: DataWriter, out: OutputStream) {
        writer.writeInt32(out, silenceLength)
    }
}
